doc/source/notebooks/interp_transition_guide.md

```diff
@@ -0,0 +1,249 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.0
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Interpolate transition guide
+
+This notebook contains three sets of demonstrations:
+
+- lower-level FITPACK replacements for `interp2d` for legacy bug-for-bug compatible `interp2d` replacements;
+- recommended replacements for `interp2d` for use in new code;
+- a demonstration of failure modes of 2D FITPACK-based linear interpolation and recommended replacements.
+
++++ 
+
+## 1. How to transition away from using  `interp2d`
+
+`interp2d` silently switches between interpolation on a 2D regular grid and interpolatin 2D scattered data. The switch is based on the lengths of the (raveled) `x`, `y`, and `z` arrays. In short, for regular grid use `RectBivariateSpline`; for scattered interpolation, use the `bisprep/bisplev` combo. Below we give examples of the literal point-for-point transition, which should preserve the `interp2d` results exactly.
+
++++ 
+
+### 1.1 `interp2d` on a regular grid
+
+We start from the (slightly modified) docstring example.
+
+```{code-cell} ipython3
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.interpolate import interp2d, RectBivariateSpline
+
+x = np.arange(-5.01, 5.01, 0.25)
+y = np.arange(-5.01, 7.51, 0.25)
+xx, yy = np.meshgrid(x, y)
+z = np.sin(xx**2 + 2.*yy**2)
+f = interp2d(x, y, z, kind='cubic')
+```
+
+This is the "regular grid" code path, because
+
+```{code-cell} ipython3
+z.size == len(x) * len(y)
+```
+
+Also, note that `x.size != y.size`:
```

Comment:
In ReST, there is a difference between double backticks for the literal code blocks and single backticks for intersphinx links. Which one would this be and how to generate both of these?

Reply:
I added a few examples in the notebook, in short it's a bit more explicit but I hope it works anyway.